api/v1/serviceinstance_utils.go

```diff
@@ -0,0 +1,16 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+func IsSharedDesiredStateChanged(newShareState *bool, oldShareState metav1.ConditionStatus) bool {
```

Comment:
bad method name (desired state might not be changed but sharing should still be handled), should be something like "shouldHandleSharing" or something similar

Reply:
changed